README.md

```diff
@@ -1 +1,58 @@
-# s3same
\ No newline at end of file
+# s3same
+`s3same` ("sesame", like ["open sesame"](https://en.wikipedia.org/wiki/Open_Sesame_(phrase))) creates unique-per-repo AWS credentials for Travis CI artifact uploading to S3 and encrypts those credentials with the repo's public key.
+
+## Installation
+
+Running the command
+```sh
+sudo pip install git+https://github.com/vokal/s3same.git
+```
+will install `s3same` (from the master branch of the source of this repo) as a command in `/usr/local/bin`.
+
+## Usage
+
+```
+$ s3same --help
+Usage: s3same [OPTIONS]
+
+Options:
+  --repo TEXT         The target repository  [required]
+  --pro               Use Travis CI Pro
+  --github TEXT       GitHub token
+  --owner TEXT        GitHub owner
+  --s3-bucket TEXT    S3 bucket for artifacts
+  --aws-region TEXT   AWS region
+  --aws-key TEXT      AWS key
+  --aws-secret TEXT   AWS secret
+  --aws-profile TEXT  AWS profile
+  --help              Show this message and exit.
+```
+
+Let's assume you've got your AWS credentials in `~/.aws/credentials`.  Go to your GitHub settings and [create a personal access token](https://github.com/settings/tokens/new) for `s3same` (the token must have the `repo` permission).  Create `~/.s3same` with:
+```
+GITHUB_TOKEN=put your token here
+S3_BUCKET=the bucket to which artifacts will be uploaded
+AWS_PROFILE=some profile
+AWS_REGION=your AWS region
+```
+If your AWS credentials are in the default profile, you can omit the `AWS_PROFILE` line.
+
+With all the credentials in place, running
+```
+$ s3same --repo some-repo-name --owner some-user --pro
+```
+will create an AWS IAM user unique to the repo (`s3same_travis__some-user__some-repo-name`), add that user to the `s3same_travis` AWS IAM group to give it the necessary permissions (defined by the `s3same_travis` policy), generate an AWS key and secret for that user, use the public key for the given repo on Travis CI to encrypt the key and secret, and print out the YAML snippet to use for artifact uploading credentials.
```

Comment:
Does the `s3same_travis` group need to be manually created in IAM before this can be used? I'm sure you've created it for our account already, but I'm thinking of anyone else who might use this tool.